### ISSUE TITLE
Error when implicit parameter is defined without an associated attribute

### ISSUE THREAD

**Issue by franklinsch** [franklinsch/flint#83]
Payable functions (annotated with `@payable`), need to have a corresponding variable marked `implicit` and of some currency type (`Ether` or `Wei`), e.g.

```swift
@payable
func foo(implicit value: Wei) {
}
```

Functions which don't carry the `@payable` tag shouldn't have any `implicit` variables. The compiler should emit an error if it is the case.
